contentcuration/contentcuration/management/commands/set_file_duration.py

```diff
@@ -0,0 +1,78 @@
+import logging as logmodule
+import os
+import shutil
+import subprocess
+import tempfile
+import time
+
+from django.core.management.base import BaseCommand
+
+from contentcuration.models import File
+from contentcuration.models import MEDIA_PRESETS
+
+logmodule.basicConfig(level=logmodule.INFO)
+logging = logmodule.getLogger('command')
+
+
+CHUNKSIZE = 10000
+
+
+def extract_duration_of_media(fpath_in):
+    result = subprocess.check_output(
+        [
+            "ffprobe",
+            "-v",
+            "error",
+            "-show_entries",
+            "format=duration",
+            "-of",
+            "default=noprint_wrappers=1:nokey=1",
+            "-loglevel",
+            "panic",
+            str(fpath_in),
+        ]
+    )
```

Comment:
In case you'd like to try it, it seems this will pipe a stream like a file object (`f_in`) directly to the command, which could remove the need for the temporary files and such
```python
    result = subprocess.check_output(
        [
            "ffprobe",
            "-v",
            "error",
            "-show_entries",
            "format=duration",
            "-of",
            "default=noprint_wrappers=1:nokey=1",
            "-loglevel",
            "panic",
            "-"
        ],
        stdin=f_in,
    )
```
The key features being the trailing `-` on the command, and the `stdin` kwarg.

Reply:
I can test it locally - our GCS Storage implementation is already generating a temporary file locally, so this was doing it twice, so would be nice to avoid that if possible.

Reply:
Worked!